File: client/src/clashmenu/tabs/BattleTabStyles.cs
/**
 * Battle Tab Styles - CSS en JavaScript
 * Styles spécifiques à l’onglet Bataille
 */
class BattleTabStyles {
  static get colors() {
    return {
      white: '#ffffff',
      gold: '#ffd700',
      green: '#4caf50',
      red: '#f44336',
      orange: '#ff9800',
      darkBlue: '#0f1419',
      royalBlue: '#1a237e',
      neonBlue: '#00bcd4',
      electric: '#00e5ff'
    };
  }

  static getCSS() {
    return this.getBattleTabStyles();
  }

  static getBattleTabStyles() {
    const { white, gold, green, red, orange, darkBlue, royalBlue, neonBlue, electric } = this.colors;

    return `
    .battle-tab {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: calc(100% - 80px);
      padding: 20px;
      overflow-y: auto;
      display: none;

      /* Masquer la scrollbar */
      scrollbar-width: none;     /* Firefox */
      -ms-overflow-style: none;  /* IE et Edge */
    }
    .battle-tab::-webkit-scrollbar { 
      display: none;             /* Chrome, Safari, Edge */
    }
    
    .battle-tab.active {
      display: block;
    }

    /* Header avec stats joueur */
    .battle-header {
      text-align: center;
      margin-bottom: 40px;
      padding: 30px 20px;
      background: linear-gradient(135deg, ${darkBlue}cc, ${royalBlue}cc);
      border-radius: 20px;
      backdrop-filter: blur(15px);
      border: 2px solid ${neonBlue}44;
    }

    .player-name {
      font-size: 32px;
      font-weight: bold;
      color: ${gold};
      margin-bottom: 10px;
      text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5);
    }

    .player-stats {
      display: flex;
      justify-content: center;
      gap: 40px;
      margin-top: 20px;
    }

    .stat-item {
      text-align: center;
    }

    .stat-value {
      font-size: 24px;
      font-weight: bold;
      color: ${white};
      display: block;
    }

    .stat-label {
      font-size: 14px;
      color: ${neonBlue};
      text-transform: uppercase;
      letter-spacing: 1px;
      margin-top: 5px;
    }

    /* Status de connexion */
    .connection-status {
      text-align: center;
      padding: 15px;
      margin-bottom: 30px;
      border-radius: 15px;
      font-weight: bold;
      transition: all 0.3s ease;
    }

    .connection-status.connected {
      background: ${green}22;
      color: ${green};
      border: 2px solid ${green}44;
    }

    .connection-status.connecting {
      background: ${orange}22;
      color: ${orange};
      border: 2px solid ${orange}44;
    }

    .connection-status.error {
      background: ${red}22;
      color: ${red};
      border: 2px solid ${red}44;
    }

    .connection-status.searching {
      background: ${electric}22;
      color: ${electric};
      border: 2px solid ${electric}44;
      animation: pulse 1.5s infinite;
    }
      
    /* Boutons de bataille */
    .battle-buttons {
      display: flex;
      flex-direction: column;
      gap: 20px;
      max-width: 400px;
      margin: 0 auto;
    }
    
    .battle-button {
      padding: 20px 30px;
      font-size: 20px;
      font-weight: bold;
      text-transform: uppercase;
      letter-spacing: 2px;
      border: none;
      border-radius: 15px;
      cursor: pointer;
      transition: all 0.3s ease;
      position: relative;
      overflow: hidden;
    }
    
    .battle-button:before {
      content: '';
      position: absolute;
      top: 0;
      left: -100%;
      width: 100%;
      height: 100%;
      background: linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.2), transparent);
      transition: left 0.5s ease;
    }
    
    .battle-button:hover:before {
      left: 100%;
    }
    
    .battle-button:hover {
      transform: translateY(-3px);
      box-shadow: 0 10px 25px rgba(0, 0, 0, 0.3);
    }
    
    .battle-button:active {
      transform: translateY(-1px);
    }
    
    .battle-button.primary {
      background: linear-gradient(135deg, ${green}, #45a049);
      color: ${white};
      box-shadow: 0 8px 20px ${green}44;
    }
    
    .battle-button.secondary {
      background: linear-gradient(135deg, ${neonBlue}, #00acc1);
      color: ${white};
      box-shadow: 0 8px 20px ${neonBlue}44;
    }
    
    .battle-button.tertiary {
      background: linear-gradient(135deg, ${orange}, #f57c00);
      color: ${white};
      box-shadow: 0 8px 20px ${orange}44;
    }
    
    .battle-button:disabled {
      opacity: 0.5;
      cursor: not-allowed;
      transform: none;
    }
    
    .battle-button:disabled:hover {
      transform: none;
      box-shadow: none;
    }
    
    /* Mode de bataille */
    .battle-modes {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
      gap: 15px;
      margin-top: 30px;
    }
    
    .battle-mode-card {
      background: linear-gradient(135deg, ${darkBlue}aa, ${royalBlue}aa);
      border: 2px solid ${neonBlue}33;
      border-radius: 15px;
      padding: 20px;
      text-align: center;
      cursor: pointer;
      transition: all 0.3s ease;
      backdrop-filter: blur(10px);
    }
    
    .battle-mode-card:hover {
      border-color: ${neonBlue};
      transform: translateY(-5px);
      box-shadow: 0 10px 30px rgba(0, 188, 212, 0.2);
    }
    
    .battle-mode-icon {
      font-size: 36px;
      margin-bottom: 10px;
      display: block;
    }
    
    .battle-mode-title {
      color: ${white};
      font-weight: bold;
      font-size: 16px;
      margin-bottom: 8px;
    }
    
    .battle-mode-desc {
      color: ${neonBlue};
      font-size: 12px;
      line-height: 1.4;
    }
    `;
  }
}

export default BattleTabStyles;

File: client/src/clashmenu/tabs/BattleTabStyles.js
class BattleTabStyles {
  static get colors() {
    return {
      white: '#ffffff',
      gold: '#ffd700',
      darkBlue: '#0f1419',
      royalBlue: '#1a237e',
      neonBlue: '#00bcd4',
      gray: '#9e9e9e'
    };
  }

  static getCSS() {
    return this.getBattleTabStyles();
  }

  static getBattleTabStyles() {
    const { white, gold, darkBlue, royalBlue, gray } = this.colors;

    return `
    .battle-tab {
      position: absolute;
      top: 60px;
      left: 0;
      width: 100%;
      height: calc(100% - 100px);
      display: flex;
      flex-direction: column;
      align-items: center;
      overflow-y: auto;
    }

    /* --- Top bar with banner --- */
    .battle-topbar {
      position: relative;
      width: 100%;
      height: 100px;
      border-radius: 0;
      margin-bottom: 15px;
      overflow: hidden;
      box-shadow: 0 4px 12px rgba(0,0,0,0.4);
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 0 15px;
    }

    .topbar-banner {
      position: absolute;
      inset: 0;
      width: 100%;
      height: 100%;
      object-fit: cover;
      cursor: pointer;
      filter: brightness(0.7);
    }

    .topbar-content {
      position: relative;
      z-index: 2;
      width: 100%;
      display: flex;
      align-items: center;
      justify-content: space-between;
    }

    /* Left side (avatar + name + trophies) */
    .topbar-left {
      display: flex;
      align-items: center;
      gap: 10px;
    }

    .player-avatar {
      width: 60px;
      height: 60px;
      border-radius: 50%;
      border: 3px solid ${gold};
      background: ${gray};
      object-fit: cover;
      cursor: pointer;
    }

    .player-info {
      display: flex;
      flex-direction: column;
    }

    .player-name {
      font-size: 16px;
      font-weight: bold;
      color: ${white};
    }

    .player-trophies {
      font-size: 14px;
      color: ${gold};
    }

    /* Right side (icons) */
    .topbar-right {
      display: flex;
      gap: 10px;
    }

    .topbar-btn {
      width: 42px;
      height: 42px;
      border-radius: 50%;
      border: none;
      background: rgba(255,255,255,0.15);
      color: ${white};
      font-size: 20px;
      cursor: pointer;
      transition: transform 0.2s ease, background 0.2s ease;
    }

    .topbar-btn:hover {
      transform: scale(1.1);
      background: rgba(255,255,255,0.25);
    }

    /* --- Arena --- */
    .arena-section {
      flex: 1;
      display: flex;
      align-items: center;
      justify-content: center;
    }

    .arena-image {
      max-width: 100%;
      max-height: 220px;
      object-fit: contain;
      border-radius: 15px;
      box-shadow: 0 8px 20px rgba(0,0,0,0.5);
    }

    /* --- Battle action --- */
    .battle-action {
      display: flex;
      align-items: center;
      justify-content: center;
      gap: 15px;
      margin: 15px 0;
    }

    .battle-main-btn {
      padding: 20px 50px;
      font-size: 22px;
      font-weight: bold;
      text-transform: uppercase;
      background: linear-gradient(135deg, ${gold}, #ffb300);
      color: ${darkBlue};
      border: none;
      border-radius: 20px;
      cursor: pointer;
      box-shadow: 0 8px 20px rgba(0,0,0,0.3);
      transition: transform 0.2s ease;
    }

    .battle-main-btn:hover {
      transform: translateY(-3px);
    }

    .battle-main-btn:active {
      transform: translateY(-1px);
    }

    .battle-mode-btn {
      width: 52px;
      height: 52px;
      border-radius: 50%;
      background: linear-gradient(135deg, ${royalBlue}, ${darkBlue});
      color: ${white};
      border: 2px solid ${gold};
      font-size: 22px;
      cursor: pointer;
      transition: transform 0.2s ease;
    }

    .battle-mode-btn:hover {
      transform: scale(1.1);
    }
    `;
  }
}

export default BattleTabStyles;
